app/code/Magento/Ui/view/base/web/js/form/form.js

```diff
@@ -339,6 +339,7 @@ define([
          */
         reset: function () {
             this.source.trigger('data.reset');
+            $('._has-datepicker').val('');
```

Comment:
Business logic must not rely on CSS classes, see [Code Demarcation Standard](https://devdocs.magento.com/guides/v2.3/coding-standards/code-standard-demarcation.html). Please add appropriate data attributes instead.

Reply:
@orlangur updated.